### ISSUE TITLE
Model Weights Not Updating During Training Due to Incorrect Model Call

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1450]
The model's weights are not being updated during training due to the use of model.predict() instead of model() and the loss function is not being properly calculated and backpropagated through the network.
